File: src/main/java/com/pyamsoft/padlock/dagger/settings/SettingsModule.java
/*
 * Copyright 2016 Peter Kenji Yamanaka
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.pyamsoft.padlock.dagger.settings;

import android.content.Context;
import android.support.annotation.NonNull;
import com.pyamsoft.padlock.R;
import com.pyamsoft.padlock.app.settings.SettingsPresenter;
import dagger.Module;
import dagger.Provides;
import javax.inject.Named;

@Module public class SettingsModule {

  private final long defaultTimeoutTime;

  public SettingsModule(final @NonNull Context context) {
    final Context appContext = context.getApplicationContext();
    defaultTimeoutTime = Long.parseLong(appContext.getString(R.string.timeout_time_default));
  }

  @Provides SettingsPresenter provideSettingsPresenter(
      final @NonNull SettingsPresenterImpl presenter) {
    return presenter;
  }

  @Provides SettingsInteractor provideSettingsInteractor(
      final @NonNull SettingsInteractorImpl interactor) {
    return interactor;
  }

  @Provides @Named("timeout_default") long provideTimeoutDefault() {
    return defaultTimeoutTime;
  }
}

File: padlock-settings/src/main/java/com/pyamsoft/padlock/settings/SettingsModule.kt
/*
 *     Copyright (C) 2017 Peter Kenji Yamanaka
 *
 *     This program is free software; you can redistribute it and/or modify
 *     it under the terms of the GNU General Public License as published by
 *     the Free Software Foundation; either version 2 of the License, or
 *     (at your option) any later version.
 *
 *     This program is distributed in the hope that it will be useful,
 *     but WITHOUT ANY WARRANTY; without even the implied warranty of
 *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *     GNU General Public License for more details.
 *
 *     You should have received a copy of the GNU General Public License along
 *     with this program; if not, write to the Free Software Foundation, Inc.,
 *     51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 */

package com.pyamsoft.padlock.settings

import android.support.annotation.CheckResult
import com.pyamsoft.padlock.base.db.PadLockDBDelete
import com.pyamsoft.padlock.base.preference.ClearPreferences
import com.pyamsoft.padlock.base.preference.InstallListenerPreferences
import com.pyamsoft.padlock.base.preference.MasterPinPreferences
import com.pyamsoft.pydroid.bus.EventBus
import com.pyamsoft.pydroid.data.Cache
import dagger.Module
import dagger.Provides
import javax.inject.Named
import javax.inject.Singleton

@Module
class SettingsModule {

  @Singleton
  @Provides
  @CheckResult internal fun provideConfirmBus(): EventBus<ConfirmEvent> =
      ConfirmEventBus()

  @Singleton
  @Provides
  @CheckResult internal fun provideSettingsInteractor(
      deleteDb: PadLockDBDelete, masterPinPreference: MasterPinPreferences,
      clearPreference: ClearPreferences, installListenerPreferences: InstallListenerPreferences,
      @Named("cache_lock_list") lockListInteractor: Cache,
      @Named("cache_lock_info") lockInfoInteractor: Cache,
      @Named("cache_lock_entry") lockEntryInteractor: Cache,
      @Named("cache_purge") purgeInteractor: Cache): SettingsInteractor {
    return SettingsInteractorImpl(deleteDb, masterPinPreference, clearPreference,
        installListenerPreferences, lockListInteractor, lockInfoInteractor, lockEntryInteractor,
        purgeInteractor)
  }
}

